core/src/main/java/io/grpc/LoadBalancer2.java

```diff
@@ -133,6 +133,20 @@ public abstract void handleResolvedAddresses(
   /**
    * Handles a state change on a Subchannel.
    *
+   * <p>The initial state of a Subchannel is IDLE. You won't get a notification for the initial IDLE
+   * state.
+   *
+   * <p>If a Subchannel has transitioned from a state that would be picked for RPCs, to a state that
+   * would not be picked, this method should create a new picker and call {@link
+   * Helper#updatePicker}, in order to prevent unnecessary of RPCs. Please refer to {@link
```

Comment:
Sorry, should be "unnecessary delayed of RPCs".

We do promptly fail fail-fast RPCs, but if `updatePicker` is not called when it's necessary, RPCs will stay buffered while they should proceed.